### PR TITLE
Naming strategy joinTableColumnName has to know if it is called from the owning or owned (inverse) context

### PR DESCRIPTION
**Pull request by risa**
PR for feature request https://github.com/typeorm/typeorm/issues/1282

Including simple tests.

### CODE REVIEW

File: src/naming-strategy/DefaultNamingStrategy.ts [hunk 1]
```diff
@@ -60,7 +60,7 @@ export class DefaultNamingStrategy implements NamingStrategyInterface {
         return columnName + "_" + index;
     }
 
-    joinTableColumnName(tableName: string, propertyName: string, columnName?: string): string {
+    joinTableColumnName(tableName: string, propertyName: string, columnName?: string, inverse?: boolean): string {
```
Comment (pleerock) [inline src/naming-strategy/DefaultNamingStrategy.ts]: Maybe its better to create separate `joinTableInverseColumnName` method instead?

File: test/github-issues/1282/issue-1282.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+        // joinTbelColumnName was called at least twice
+        expect(namingStrategy.calls.length).greaterThan(1);
+
+        // the first call was with inverse=false - i.e. for the owning part of the relation
+        expect(namingStrategy.calls[0]).to.be.false;
+
+        // the second call was with invers=true - i.e. for the owned part of the relation
+        expect(namingStrategy.calls[1]).to.be.true;
+
```
Comment (pleerock) [inline test/github-issues/1282/issue-1282.ts]: probably you can check how your database column are named in the database this way:

```ts
connection.getManyToManyMetadata(Animal, "categories")
     .columns.find(column => column.databaseName === "...")
     .should.not.be.undefined;
```

File: src/naming-strategy/DefaultNamingStrategy.ts (hunk 1 above)
Comment (risa) [reply src/naming-strategy/DefaultNamingStrategy.ts]: That would definitely be more transparent, I just did not want to introduce new methods. But if that is OK with you then I am going to change it.

File: test/github-issues/1282/issue-1282.ts (hunk 2 above)
Comment (risa) [reply test/github-issues/1282/issue-1282.ts]: Looks better, I will give it a try. Thanks.

File: src/naming-strategy/DefaultNamingStrategy.ts (hunk 1 above)
Comment (pleerock) [reply src/naming-strategy/DefaultNamingStrategy.ts]: yeah lets change it

**Comment by risa** [comment]
@pleerock, sorry for the delay - I have updated the PR to include both functions as you suggested:
- ```joinTableColumnName``` and
- ```joinTableInverseColumnName```
Could you please have a look at the PR? 

Thanks


Richard

**Comment by pleerock** [comment]
thank you, what about this my comment:

> probably you can check how your database column are named in the database this way:
```ts
connection.getManyToManyMetadata(Animal, "categories")
     .columns.find(column => column.databaseName === "...")
     .should.not.be.undefined;
```

**Comment by risa** [comment]
Sorry, I forgot about this one, I will try it later today ...

**Comment by risa** [comment]
Updated the testing logic. Unfortunately I could not make the call as elegant as you proposed because there is strictNullChecks=true parameter in tsconfig.json and the getEntityMetadata returns a potentially undefined value, so typescript compiler kept bothering me until I made the explicit assignment with if. If you know a more elegant way please leave me a comment. Thanks!

**Comment by pleerock** [comment]
That's fine, thank you very much!

**Comment by pleerock** [comment]
Will be released in 0.1.8 probably next week, but let me know if you need it super urgently

**Pull request merged**